sdk/create-dapp/templates/react-e2e-counter/src/Counter.tsx

```diff
@@ -0,0 +1,90 @@
+import {
+  useCurrentAccount,
+  useSignAndExecuteTransactionBlock,
+  useSuiClientQuery,
+} from "@mysten/dapp-kit";
+import { SuiObjectData } from "@mysten/sui.js/client";
+import { TransactionBlock } from "@mysten/sui.js/transactions";
+import { Button, Flex, Heading, Text } from "@radix-ui/themes";
+import { PACKAGE_ID } from "./constants";
+
+export function Counter({ id }: { id: string }) {
+  const currentAccount = useCurrentAccount();
+  const { mutateAsync: signAndExecute } = useSignAndExecuteTransactionBlock({});
```

Comment:
nit: I'd slightly prefer if we used `mutate` with the `onSuccess` callback over using `mutateAsync` because it makes the code easier to read

https://tkdodo.eu/blog/mastering-mutations-in-react-query#mutate-or-mutateasync